From bd360f5c1c35199106cd97a6d21fdd3af7c08a04 Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 22 Jul 2014 14:40:00 -0400
Subject: [virt] kvm/x86: drop set_rflags callback

Message-id: <1406040016-3289-206-git-send-email-pbonzini@redhat.com>
Patchwork-id: 86284
O-Subject: [RHEL7 PATCH v2 205/221] KVM: x86: drop set_rflags callback
Bugzilla: 1078775 1088784 1116936
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Not needed anymore now that the CPL is computed directly
during task switch.

Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
(cherry picked from commit fb5e336b977086557739791ed51955c5913dc773)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/include/asm/kvm_emulate.h b/arch/x86/include/asm/kvm_emulate.h
index 24ec121..a04fe4e 100644
--- a/arch/x86/include/asm/kvm_emulate.h
+++ b/arch/x86/include/asm/kvm_emulate.h
@@ -189,7 +189,6 @@ struct x86_emulate_ops {
  void (*set_idt)(struct x86_emulate_ctxt *ctxt, struct desc_ptr *dt);
  ulong (*get_cr)(struct x86_emulate_ctxt *ctxt, int cr);
  int (*set_cr)(struct x86_emulate_ctxt *ctxt, int cr, ulong val);
- void (*set_rflags)(struct x86_emulate_ctxt *ctxt, ulong val);
  int (*cpl)(struct x86_emulate_ctxt *ctxt);
  int (*get_dr)(struct x86_emulate_ctxt *ctxt, int dr, ulong *dest);
  int (*set_dr)(struct x86_emulate_ctxt *ctxt, int dr, ulong value);
diff --git a/arch/x86/kvm/x86.c b/arch/x86/kvm/x86.c
index 001563f..454f96d 100644
--- a/arch/x86/kvm/x86.c
+++ b/arch/x86/kvm/x86.c
@@ -4640,11 +4640,6 @@ static int emulator_set_cr(struct x86_emulate_ctxt *ctxt, int cr, ulong val)
  return res;
 }
 
-static void emulator_set_rflags(struct x86_emulate_ctxt *ctxt, ulong val)
-{
- kvm_set_rflags(emul_to_vcpu(ctxt), val);
-}
-
 static int emulator_get_cpl(struct x86_emulate_ctxt *ctxt)
 {
  return kvm_x86_ops->get_cpl(emul_to_vcpu(ctxt));
@@ -4829,7 +4824,6 @@ static const struct x86_emulate_ops emulate_ops = {
  .set_idt      = emulator_set_idt,
  .get_cr              = emulator_get_cr,
  .set_cr              = emulator_set_cr,
- .set_rflags          = emulator_set_rflags,
  .cpl                 = emulator_get_cpl,
  .get_dr              = emulator_get_dr,
  .set_dr              = emulator_set_dr,
-- 
1.7.1